src/jsep.js

```diff
@@ -658,6 +697,54 @@ jsep.toString = function() {
 	return 'JavaScript Expression Parser (JSEP) v' + jsep.version;
 };
 
+jsep.hooks = {
```

Comment:
We should have a separate module for the hooks object that jsep imports, since we're using a build tool anyway.
You could even take the Bliss one wholesale.

Reply:
I'm not that familiar with rollup and not sure what you want the final output to look like. Do you want to build a separate set of output files for each plugin? Or do you want it all bundled together? Or maybe you could make an update or branch with a build system update with proper directory structure and I could split this PR from there?

Reply:
The bundles in `dist/` would definitely include any JS in `src`, so if we have a `hooks.js` that will be included. But as to which plugins to include ...excellent question. I suppose we should have a default list of plugins that is included in the default bundles, e.g. ternary, and anyone that wishes to do something different can do manual imports or a custom build. Alternatively the default bundles could be without any plugins, or we could have multiple bundles with different sets of plugins (e.g. none, default, all). What do you think?

Reply:
We could potentially set up multiple npm packages (i.e. one per plugin). But these plugins are all so tiny, I think everything could be built together (and ESM will allow tree shaking anyway)?
So maybe:
```
jsep
├── dist
├── src
│   ├── jsep.js
│   ├── jsep.test.js
│   ├── hooks.js
│   ├── plugins
│   │   ├── ternary
│   │   │   ├── ternary.js
│   │   │   ├── ternary.test.js
│   │   ├── ...
├── test (maybe still a test folder, with esprima)?
├── typings
└── package.json, build files...
```

It would make it simpler for including ternary by default... thoughts?

Reply:
> But these plugins are all so tiny, I think everything could be built together (and ESM will allow tree shaking anyway)?

Yup.

I like the proposed dir structure, the only thing I'm not sure about is whether `plugins/` should be in `src/` or adjacent to it. Might be worth it to see what a sample of popular projects do, but I don't have a strong opinion either way.